lib/pcr.c

```diff
@@ -305,6 +305,18 @@ bool pcr_get_banks(ESYS_CONTEXT *esys_context, TPMS_CAPABILITY_DATA *capability_
     *capability_data = *capdata_ret;
 
     unsigned i;
+
+    // If the TPM support more bank algorithm that we currently
+    // able to manage, throw an error
+    if (capability_data->data.assignedPCR.count > sizeof(algs->alg)) {
+        LOG_ERR("Current implementation do not support more than %ld banks "
+                "(%d banks supported by TPM)", 
```

Comment:
format specifiers wrong, should be %zu and (%zu or PRIu32).

Grammar, "Current implementation does not support more than %zu banks, got: "PRIu32"";